test/uk/gov/hmrc/personaldetailsvalidation/PersonalDetailsValidationResourceControllerSpec.scala

```diff
@@ -18,141 +18,170 @@ package uk.gov.hmrc.personaldetailsvalidation
+
     "return Not Found http status code if repository does not return personal details validation" in new Setup {
-      (mockRepository.get(_: ValidationId)(_: ExecutionContext)) expects(personalDetailsValidationId, instanceOf[MdcLoggingExecutionContext]) returns Future.successful(None)
+      val validationId = ValidationId()
+
+      (mockRepository.get(_: ValidationId)(_: ExecutionContext))
+        .expects(validationId, instanceOf[MdcLoggingExecutionContext])
+        .returns(Future.successful(None))
 
-      val response = controller.get(personalDetailsValidationId)(request).futureValue
+      val response = controller.get(validationId)(request).futureValue
 
       status(response) shouldBe NOT_FOUND
     }
 
     "return OK http status code if repository returns personal details validation" in new Setup {
-      (mockRepository.get(_: ValidationId)(_: ExecutionContext)) expects(personalDetailsValidationId, instanceOf[MdcLoggingExecutionContext]) returns Future.successful(Some(personalDetailsValidation))
+      forAll { personalDetailsValidation: PersonalDetailsValidation =>
+        (mockRepository.get(_: ValidationId)(_: ExecutionContext))
+          .expects(personalDetailsValidation.id, instanceOf[MdcLoggingExecutionContext])
+          .returns(Future.successful(Some(personalDetailsValidation)))
 
-      val response = controller.get(personalDetailsValidationId)(request).futureValue
+        val response = controller.get(personalDetailsValidation.id)(request).futureValue
 
-      status(response) shouldBe OK
+        status(response) shouldBe OK
+      }
     }
 
     "return personalDetailsValidation id in response body" in new Setup {
-      (mockRepository.get(_: ValidationId)(_: ExecutionContext)) expects(personalDetailsValidationId, instanceOf[MdcLoggingExecutionContext]) returns Future.successful(Some(personalDetailsValidation))
+      forAll { personalDetailsValidation: PersonalDetailsValidation =>
+        (mockRepository.get(_: ValidationId)(_: ExecutionContext))
+          .expects(personalDetailsValidation.id, instanceOf[MdcLoggingExecutionContext])
+          .returns(Future.successful(Some(personalDetailsValidation)))
 
-      val response = controller.get(personalDetailsValidationId)(request).futureValue
+        val response = controller.get(personalDetailsValidation.id)(request).futureValue
 
-      (jsonBodyOf(response) \ "id").as[String] shouldBe personalDetailsValidation.id.value.toString
+        (jsonBodyOf(response) \ "id").as[String] shouldBe personalDetailsValidation.id.value.toString
+      }
     }
 
-    "return personal details in response body" in new Setup {
-      (mockRepository.get(_: ValidationId)(_: ExecutionContext)) expects(personalDetailsValidationId, instanceOf[MdcLoggingExecutionContext]) returns Future.successful(Some(personalDetailsValidation))
+    "return personal details in response body for SuccessfulPersonalDetailsValidation" in new Setup {
```

Comment:
Add a test to verify personal details are not returned for FailedPersonalDetailValidation